javaparser-core/src/main/java/com/github/javaparser/ast/body/ModifierSet.java

```diff
@@ -19,8 +19,11 @@
 package com.github.javaparser.ast.body;
 
 import com.github.javaparser.ast.AccessSpecifier;
+import com.github.javaparser.ast.Modifiers;
+import com.github.javaparser.ast.NodeWithModifiers;
 
 import java.lang.reflect.Modifier;
```

Comment:
@ftomassetti j.l.r is another view from what we have to take.
If you look at the spec (JLS8 Chapter 9.4 p. 286)
> InterfaceMethodModifier: one of
>    *Annotation* **public** **abstract** **default** **static** **strictfp**

Thus I think we have to have our own constant. For the 'default', in my fix https://github.com/ptitjes/javaparser/commit/cc6378936b0a072b80b1d9d420b66b8337cbc033#diff-af73e9494c4ad27e96149400188c9ea3R56, I had to add a new constant not present in j.l.r.Modifier.

I understand that we need to keep the integer values for backward compatibility. And this something enums are good for! So I restate it:
```
public interface NodeWithModifiers {

	@Deprecated
	int getModifiers();

	@Deprecated
	void setModifiers(int modifiers);

	EnumSet<Modifier> getModifiersSet();
}

public enum Modifier {

	// Exact values taken from j.l.r.Modifier
	// We can remove the integer field completely after deprecation
	PUBLIC(0x00000001, "public"),
	PRIVATE(0x00000002, "private"),
	PROTECTED(0x00000004, "protected"),
	STATIC(0x00000008, "static"),

	// ...

	STRICTFP(0x00000800, "strictfp"),
	// Note that the last comma after the last constant IS valid java
	// This way of defining enums is diff-proof when you add new values
	;

	private final int integerValue;
	private final String label;

	Modifier(int integerValue, String label) {
		this.integerValue = integerValue;
		this.label = label;
	}

	@Deprecated
	public int integerValue() {
		return integerValue;
	}

	public String label() {
		return label;
	}

	// We can still add helpers
	public static boolean isPublic(NodeWithModifiers node) {
		return node.getModifiersSet().contains(PUBLIC);
	}

	public static void setPublic(NodeWithModifiers node) {
		node.getModifiersSet().add(PUBLIC);
	}

	// ...
}

@Deprecated
public /**/ class ModifierSet {

	// For compatibility
	@Deprecated
	public static final int PUBLIC = Modifier.PUBLIC.integerValue();

	@Deprecated
	public static final int PRIVATE = Modifier.PRIVATE.integerValue();

	// ...

	@Deprecated
	public static int addModifier(int modifiers, int mod) {
		return modifiers | mod;
	}

	@Deprecated
	public static int removeModifier(int modifiers, int mod) {
		return modifiers & ~mod;
	}

	@Deprecated
	public static boolean hasModifier(int modifiers, int modifier) {
		return (modifiers & modifier) != 0;
	}

	@Deprecated
	public static boolean isPublic(int modifiers) {
		return (modifiers & PUBLIC) != 0;
	}

	// ...
}
```

Let's compare the now deprecated way and the new way:
```
NodeWithModifiers method = /* ... */;

// Querying modifiers
{
	// Deprecated way
	if (ModifierSet.isPublic(method.getModifiers())) { /* ... */ }
}
{
	// New way
	if (method.getModifiersSet().contains(Modifier.PUBLIC)) { /* ... */ }
	// Note nice completion of ^^^ in IDEs
	// Whereas completion is useless with integers
}

// Updating modifiers
{
	// Deprecated way
	int modifiers = method.getModifiers();
	int modifiers2 = ModifierSet.removeModifier(modifiers, ModifierSet.PRIVATE);
	int modifiers3 = ModifierSet.addModifier(modifiers2, ModifierSet.PUBLIC);
	method.setModifiers(modifiers3);
}
{
	// New way
	EnumSet<Modifier> modifiers = method.getModifiersSet();
	modifiers.add(Modifier.PRIVATE);
	modifiers.add(Modifier.PUBLIC);
	// Note nice completion of ^^^ in IDEs
	// Whereas completion is useless with integers
}
```

IMHO it is a complete win-win to go that way while still supporting the old way until end of deprecation time.